Compiler/preproc/preprocessor.cpp

```diff
@@ -0,0 +1,429 @@
+#include "preproc/preprocessor.h"
+#include "util/textstreamwriter.h"
+#include "util/textreader.h"
+#include "script/cc_error.h"
+
+#define STRINGIFY2(X) #X
+#define STRINGIFY(X) STRINGIFY2(X)
+
+using namespace AGS::Common;
+
+extern int currentline; // in script/script_common
+
+namespace AGS {
+namespace Preprocessor {
+
+    static const size_t NOT_FOUND = -1;
+
+#if AGS_PLATFORM_OS_WINDOWS
+    static const char * li_end = "\r\n";
+#else
+    static const char * li_end = "\n";
+#endif
+
+    class StringBuilder : TextWriter {
+    private:
+        String _str = nullptr;
+    public:
+        explicit StringBuilder(size_t size) {
+            _str.Reserve(size);
+        }
+
+        void WriteLine(const String& str) override { _str.Append(str); _str.Append(li_end); }
+        bool IsValid() const override { return true; }
+        void WriteChar(char c) override {  _str.AppendChar(c); }
+        void WriteString(const String &str) override { _str.Append(str);}
+        void WriteFormat(const char *fmt, ...) override { va_list argptr; _str.Append(String::FromFormatV(fmt, argptr)); }
```

Comment:
This is not going to work, you declare `argptr` but not initialize it.

For the reference, this is how it's done: https://github.com/adventuregamestudio/ags/blob/008b2d2c61420f6f2c996317a0eb56a49e708095/Common/util/string.cpp#L650

Reply:
thanks, I didn't knew that. Just fixed.